code/src/java/pcgen/cdom/facet/base/AbstractItemFacet.java

```diff
@@ -110,9 +110,23 @@ public T remove(IDT id)
 	 * @return the item value for this AbstractItemFacet and the Player
 	 *         Character represented by the given PCGenIdentifier.
 	 */
-	public T get(IDT id)
+	public T get(final IDT id)
+	{
+		final T tmp = (T)getCache(id);
```

Comment:
I'm sorry, but these final entries aren't valuable.  There are plenty of automated tools that want to do this, but this destroys code clarity.
Final should be used where it actually provides a hint to the user, not everywhere it might be possible to use it under any circumstances.
These usages as well as on parameters I think are actually bad usages of final